src/index.js

```diff
@@ -558,20 +558,19 @@ Adapter.extend({
    * @param {Object} config.headers Headers for the request.
    * @param {Object} config.params Querystring for the request.
    * @param {string} config.url Url for the request.
-   * @param {Object} [opts] Configuration options.
    */
-  fetch (config, opts) {
+  fetch (config) {
     const requestConfig = {
       method: config.method,
       // turn the plain headers object into the Fetch Headers object
-      headers: new Headers(config.headers)
+      headers: new Headers(config.headers || {})
     }
 
     if (config.data) {
       requestConfig.body = utils.toJson(config.data)
     }
 
-    return fetch(new Request(buildUrl(config.url, config.params), requestConfig))
+    return fetch(buildUrl(config.url, config.params), requestConfig)
```

Comment:
According to the docs argument signature for fetch and Request is equivalent https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch, so it shouldn't need to explicitly create a `new Request()`.